src/witness_generator/machines/fixed_lookup_machine.rs

```diff
@@ -58,6 +59,15 @@ impl Machine for FixedLookup {
             return Some(Ok(vec![]));
         }
 
+        // get the values of the fixed columns
```

Comment:
Could you move this block to line 52? I think we should first fully decide if we are a matching machine before we make any statements.